.github/workflows/pre-commit-ubuntu-18.04.yaml

```diff
@@ -0,0 +1,279 @@
+      run:
+        shell: bash
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04]
+        platform: [x64]
+
+    container:
+      image: quay.io/asenyaev/opencv-ubuntu:18.04
+      volumes:
+        - ${{ github.workspace }}:/opencv
+
+    steps:
+    - name: Checkout opencv
+      uses: actions/checkout@v3
+      with:
+        submodules: false
+        fetch-depth: 0
+
+    - name: Clone opencv_contrib
+      run: git clone --single-branch --branch ${{ github.base_ref }} --depth 1 https://github.com/opencv/opencv_contrib.git /opencv_contrib
+
+    - name: Build OpenCV Contrib
+      run: |
+        cd /opencv-contrib-build
+        cmake -G Ninja $EXTRA_CMAKE_OPTIONS -DOPENCV_EXTRA_MODULES_PATH=/opencv_contrib/modules /opencv
+        ninja
+
+
+  opencv-test:
+    needs: [build-opencv]
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04]
+        platform: [x64]
+        test-name:
+          [
+            "opencv_test_3d",
+            "opencv_test_calib",
+            "opencv_test_core",
+            "opencv_test_dnn",
+            "opencv_test_features2d",
+            "opencv_test_flann",
+            "opencv_test_gapi",
+            "opencv_test_highgui",
+            "opencv_test_imgcodecs",
+            "opencv_test_imgproc",
+            "opencv_test_ml",
+            "opencv_test_objdetect",
+            "opencv_test_photo",
+            "opencv_test_stereo",
+            "opencv_test_stitching",
+            "opencv_test_video",
+            "opencv_test_videoio",
+            "opencv_perf_3d",
+            "opencv_perf_calib",
+            "opencv_perf_core",
+            "opencv_perf_dnn",
+            "opencv_perf_features2d",
+            "opencv_perf_gapi",
+            "opencv_perf_imgcodecs",
+            "opencv_perf_imgproc",
+            "opencv_perf_objdetect",
+            "opencv_perf_photo",
+            "opencv_perf_stereo",
+            "opencv_perf_stitching",
+            "opencv_perf_video",
+            "opencv_perf_videoio",
+          ]
+
+    container:
+      image: quay.io/asenyaev/opencv-ubuntu:18.04
+      volumes:
+        - ${{ github.workspace }}:/opencv
+
+    steps:
+    - name: Checkout opencv
+      uses: actions/checkout@v3
+      with:
+        submodules: false
+        fetch-depth: 0
+
+    - name: Clone opencv_extra
+      run: git clone --single-branch --branch ${{ github.base_ref }} --depth 1 https://github.com/opencv/opencv_extra.git /opencv_extra
+
+    - name: Download artifacts
+      uses: actions/download-artifact@v2
+      with:
+        name: opencv-build
+        path: /opencv-build
+
+    - name: Test OpenCV - ${{ matrix.test-name }}
+      run: |
+        cd /opencv-build
+        # in case of issue: https://github.com/actions/upload-artifact/issues/38
+        chmod +x ./bin/*
+        xvfb-run -a ./bin/${{ matrix.test-name }}
```

Comment:
`_perf_` test should be executed in accuracy mode: `--perf_force_samples --perf_verify_sanity`

Reply:
Okay, I'll add these flags.